src/smoke-test/java/no.difi.sdp.client2.smoketest/SikkerDigitalPostKlientST.java

```diff
@@ -0,0 +1,186 @@
+                "1) Sett environmentvariabel '" + VIRKSOMHETSSERTIFIKAT_STI + "' til full sti til virksomhetsssertifikatet. \n" +
+                "2) Sett environmentvariabel '" + VIRKSOMHETSSERTIFIKAT_ALIAS + "' til aliaset (siste avsnitt, første del før komma): \n" +
+                "       keytool -list -keystore VIRKSOMHETSSERTIFIKAT.p12 -storetype pkcs12 \n"+
+                "3) Sett environmentvariabel '" + VIRKSOMHETSSERTIFIKAT_PASSORD + "' til passordet til virksomhetssertifikatet. \n";
+
+        if(value == null){
+            throw new RuntimeException(String.format("Finner ikke %s til virksomhetssertifikat. \n %s", variabel, oppsett));
+        }
+    }
+
+    private static Noekkelpar avsenderNoekkelpar() {
+        if(virksomhetssertifikatPassordValue == null){
+            throw new RuntimeException(
+                                        "Klarte ikke hente ut system env variabelen 'smoketest_passphrase'.\n "+
+                                        "Sett sertifikatpassordet i en env variabel: \n" +
+                                        "   export smoketest_passphrase=PASSPHRASE \n" +
+                                        "Hvis du debugger må env_variabel settes i test run configuration."
+            );
+        }
+
+        return Noekkelpar.fraKeyStoreUtenTrustStore(keyStore, virksomhetssertifikatAliasValue, virksomhetssertifikatPassordValue);
+    }
+
+    private static String getOrgNumberFromCertificate(){
+
+
+        try {
+            X509Certificate cert = (X509Certificate) keyStore.getCertificate(virksomhetssertifikatAliasValue);
+            if(cert == null){
+                throw new RuntimeException(String.format("Klarte ikke hente ut virksomhetssertifikatet fra keystoren med alias '%s'", virksomhetssertifikatAliasValue));
+            }
+            X500Name x500name = new JcaX509CertificateHolder(cert).getSubject();
+            RDN serialnumber = x500name.getRDNs(BCStyle.SN)[0];
+            return IETFUtils.valueToString(serialnumber.getFirst().getValue());
+        } catch (CertificateEncodingException e) {
+            throw new RuntimeException("Klarte ikke hente ut organisasjonsnummer fra sertifikatet.",e);
+        } catch (KeyStoreException e) {
+            throw new RuntimeException("Klarte ikke hente ut virksomhetssertifikatet fra keystoren.",e);
+        }
+    }
+
+    private static KeyStore getVirksomhetssertifikat(){
+
+        try {
+            keyStore = KeyStore.getInstance("PKCS12");
+            keyStore.load(new FileInputStream(virksomhetssertifikatStiValue), virksomhetssertifikatPassordValue.toCharArray());
+            return keyStore;
+        }
+        catch (Exception e) {
+            throw new RuntimeException("Kunne ikke initiere keystoren. Legg virksomhetssertifikatet.p12 i src/smoke-test/resources/. ", e);
+        }
+    }
+
+    @Test
+    public void A_send_digital_forsendelse() {
```

Comment:
Det er en workaround for å få Junit til å kjøre test A_* før test B_*. 
(Klassen har @FixMethodOrder(MethodSorters.NAME_ASCENDING)) 

Finnes det bedre måter å gjøre dette på please do tell :)